File: Java/238. Product of Array Except Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int left = 1;
        int[] res = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {            
            res[i] = left;
            left *= nums[i];
        }
        left = 1;
        for (int i = nums.length - 1; i > -1; i--) {
            res[i] *= left;
            left *= nums[i];
        }
        return res;
    }
}

File: Python/238. Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1] * len(nums)
        left = 1
        for i in range(len(nums)):
            ans[i] *= left
            left *= nums[i]
        right = 1
        for i in range(len(nums) - 1, -1, -1):
            ans[i] *= right
            right *= nums[i]
        return ans
